src/ActivityLogServiceProvider.php

```diff
@@ -11,4 +11,19 @@ class ActivityLogServiceProvider extends ServiceProvider
     protected $vendorName = 'backpack';
     protected $packageName = 'activity-log';
     protected $commands = [];
+
+    /**
+     * Register any application services.
+     *
+     * @return void
+     */
+    public function register()
+    {
+        // set backpack authguard
+        $this->booted(function () {
+            if (! config('activitylog.default_auth_driver')) {
+                config(['activitylog.default_auth_driver' => backpack_guard_name()]);
+            }
+        });
```

Comment:
![red-flag-nascar-gif-by-richard-childress-racing](https://github.com/Laravel-Backpack/activity-log/assets/1032474/5ebdb6ed-3c5a-4728-b8ff-ac2d1657a99a)

So wait... 

1. If you install ActivityLog... then the default auth driver of activity log become Backpack's? Why? 

2. Say the app the app has both users and admins (using different auth drivers). What does this do, it logs info only when admins do stuff?

3. Say the app the app has both users and admins (using different auth drivers). The app already existed, and was using Spatie/ActivityLog. Wouldn't changing the config of the spatie package on-the-fly be unexpected and dangerous? What could this cause?

Reply:
Humm, yes true!
But how could we do this? Because if we don't set the auth driver, Activity Log will not figure out the backpack guard and therefore the Causer...
Should we set it at some Backpack middleware?

Reply:
Previously I was setting the Causer on our Trait;

![image](https://github.com/Laravel-Backpack/activity-log/assets/1838187/2513b8f9-cec0-4fb7-8e25-17bc2e58abe1)

Reply:
Ouch! This is... tricky. Damn.

I'm wondering, do we have any way of getting "_the currently active guard_"? I don't think so. Cause there can be multiple guard used at the same time. So it's a choice... do you use `backpack_user()` or `Auth::user()`... hmm...

From what I understand, your previous solution of doing `backpack_user() ?? Auth::user()` in the model was superior to this one. Because:
- it also had a fallback - if the backpack guard wasn't used, it used Auth;
- I believe it also worked ok when activities were triggered from the front-end (users who are not logged into Backpack);

This is indeed tricky... @pxpm do you have any bright ideas here?

Reply:
I changed the behaviour because I was forcing the Causer. What if the causer was not an user?

Reply:
Ok turns out I was testing on the backpack website, which was using the default guard 🤦‍♂️ Me stoopid.

Reply:
Let's move this one to https://github.com/Laravel-Backpack/activity-log/issues/17 👌